### PR TITLE
[eclipse/xtext-eclipse#520] code mining support with own api

### PR DESCRIPTION
**Pull request by TraneSlink**
Signed-off-by: René Purrio <rpurrio@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend [hunk 1]
```diff
@@ -0,0 +1,169 @@
+/*******************************************************************************
+ * Copyright (c) 2018 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend]: not exported in manifest?

Comment (kthoms) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend]: make it x-internal first?

Comment (cdietrich) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend]: yes

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+		
+		public class �codeMiningStrategyClass.lastSegment� extends XtextCodeMiningProvider {
+			def override protected void createLineHeaderCodeMinings(IDocument document, XtextResource resource, CancelIndicator indicator, IAcceptor<ICodeMining> acceptor) throws BadLocationException{
+				
+				//TODO: implement me
+				//use acceptor.accept(super.createNewLineHeaderCodeMining(...)) to add a new code mining to the final list
+				//use indicator.isCanceled() to check, if a new code mining was started (and then to cancel this code mining with return)
+				
+				//example:
+				acceptor.accept(createNewLineHeaderCodeMining(1, document, "HeaderText"))
```
Comment (kthoms) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend]: comment out

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+			}
+			
+			def override protected void createLineContentCodeMinings(IDocument document, XtextResource resource, CancelIndicator indicator, IAcceptor<ICodeMining> acceptor)  throws BadLocationException {
+				
+				//TODO: implement me
+				//use acceptor.accept(super.createNewLineContentCodeMining(...)) to add a new code mining to the final list
+				//use indicator.isCanceled() to check, if a new code mining was started (and then to cancel this code mining with return)
+				
+				//example:
+				acceptor.accept(createNewLineContentCodeMining(30, "inline Annotation "))
```
Comment (kthoms) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend]: comment out

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend]: yes. i wonder why this works at all in domainmodel

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+		}
+	}
+	
+	def protected QualifiedName getCodeMiningStrategyClass () {
+		(grammar.genericIdeBasePackage+".codemining."+grammar.simpleName+"CodeMiningStrategy").toQualifiedName
+	}
+	
+	def protected generateXtendCodeMiningStrategy() {
+		fileAccessFactory.createXtendFile(codeMiningStrategyClass.toString.typeRef, '''
+		import org.eclipse.jface.text.BadLocationException
```
Comment (kthoms) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/codemining/CodeMiningFragment.xtend]: indentation

**Comment by kthoms** [comment]
Replaced by #660

**Pull request closed without merging**